text/0005-stage-transforms.md

```diff
@@ -0,0 +1,122 @@
+	e.target.x = local.x;
+	e.target.y = local.y;
+}
+```
+
+INTERNAL SOLUTION:<br> 
+It would be better to bring this type of solution into CreateJS so developers get what they expect.  Here is a test page using modified createjs code.  The fixes will be added to a pull request after discussed here.
+
+https://zimjs.com/createjs/retina2.html
+
+The blue circle shows a correct cursor rollover and can be dragged to follow the mouse using an unmodified event object stageX and stageY.  The red circle is following rawX and rawY.  The circles also work with stagemousemove and when the stage is rotated (commented out in the code). 
+
+The rectangles show localToGlobal, globalToLocal and localToLocal all working as expected in a container.  The container was scaled and rotated and cursor checked as well - all working.  Skew and multiple containers were not tested.
+
+CREATEJS CHANGES:<br> 
+Example changes can be found in here: https://zimjs.com/cdn/1.3.0/createjs_doc.js
+
+The general technique to fix these internally was provided in an existing comment in CreateJS on the Stage class on line 8707.  This is an invert() on a transform:
+
+```
+// TODO: account for stage transformations?
+// this._mtx = this.getConcatenatedMatrix(this._mtx).invert();
+// var pt = this._mtx.transformPoint(o.x, o.y);
+var evt = new createjs.MouseEvent(type, bubbles, false, pt.x, pt.y, etc.
+target.dispatchEvent(evt);
+```
+
+Instead of the change being made here, this technique was used in the following places in the modified CreateJS file with some being an invert() and some being a regular transform() adjustment:
+
+```
+// line 6697 - localToGlobal (invert transform)
+// line 6726 - globalToLocal (transform)
+// line 6866 - hitTest (transform) - not being used see _getObjectsUnderPoint
+// line 7761 - _getObjectsUnderPoint (transform)
+// line 7784 - _getObjectsUnderPoint - small adjust
+// line 8518 - _updatePointerPosition (invert transform)
+```
+
+A property on the createjs namespace was made on line 40:
+
+```
+createjs.stageTransformable = true;
+```
+
+Setting this to false will revert back to earlier CreateJS functionality. This property can also be read by code using createjs to adjust accordingly.  For example, in ZIM, we have already adjusted for stage scale.  We will now read this property and if it is true, we will not adjust for stage scale otherwise there would be a double adjustment.
+
+## How we teach this
+
+The stageTransformable property would go into the documentation along with an explanation.  Various forum questions or reports could be answered accordingly.  Otherwise, people would expect CreateJS to work in this manner so there is no need for teaching in general.
+
+## Drawbacks
+
+This will break existing code but can be turned off to not break existing code.  To update to a new version of CreateJS but keep the existing code working, the stageTransformable could be set to false:
```

Comment:
To be clear about this, would it be correct to say that it will only break existing code that fixes the issue in a different way?  If people are unaware of the issue or don't scale the stage, then they won't be affected at all.